### DISCUSSION TITLE
Roadmap: Extensible / Reusable Blocks UI

### DISCUSSION THREAD

**Discussion by jmikrut** [payloadcms/payload#288] [Roadmap]
### Description

Modularize all `block` React components into an extremely simple and reusable API so that developers can easily build replacement components on a block by block basis. For example, many "layout builder" style `block` fields might feature a `paragraph` or `richText`-style block type, where the UI of the admin panel editing experience could be similar to Medium's or WP Gutenberg's. To allow for this, we need to separate out all block components such as reordering, drag + drop, and block naming into separate and reusable React components, where the developer can then freely create their own block UI patterns.

This feature will be extremely powerful and could unlock a block UI marketplace, where developers could create highly polished blocks for commonly needed content models like paragraphs, accordions, columns, and more.   

### Implementation Detail

- Allow for developers to specify admin components for each of their `block`s
- Slightly refactor and modularize all existing React block components for easy re-use
- Potentially update the design of the Blocks field in general 

### Potential Breaking Changes

None

### Effort

High

**Comment by GermanJablo** [comment]
Closing because I understand this has been satisfactorily resolved! (correct me if you see anything missing)
